### ISSUE TITLE
Am i the only one with numpad problems?

### ISSUE THREAD

**Issue by i30817** [libretro/RetroArch#3961]
Seems like a minefield of input bugs here. I'm seeing them because i'm trying to use it since i destroyed my laptop keyboard over time, to the point that two of the normal directional keys don't even work.

Let's start with the obvious:
1. directional keys substitutes on the numpad don't have any effect on retroarch.
2. Currently, any input setup for user 1 is not working on the menu from the keyboard, except the default keys for movement and enter. X/Z do nothing. I believe this is a very recent bug, since the last time i updated from the ppa (i even deleted the key config files from ~/.config/retroarch)

then the less obvious

3. to remap a numpad key in retroarch you have to use numlock before. This is my least favorite way of configuring keys in emulators on the numpad. I get it that for consistency with the normal keyboard behavior shift+key or keylock+key do different things, but in emulators it is a hassle that doesn't gain you much of anything except a bit more key combinations you won't use. I much prefer that the keylock and shift status are ignored and the keys are also not 'confused' by their equivalents elsewhere on the keyboard (so maybe instead of ignored, treated as always on would be better), in almost all emulators (dosbox being the exception for obvious reasons). 

4. even if you do 3) currently you can't remap certain keys here. For example, numlock_8 remaps but numlock_9 doesn't waaaat?

There are probably other bugs for you to find. Is there even a way to unit test this?

**Comment by lasers** [comment]
No more X/Y.

**Comment by i30817** [comment]
Yeah, that's a terrible decision then. I don't even know why you wouldn't want the consistency.

**Comment by lasers** [comment]
It messes with the keyboard for long time in prompts. I used to not be able to type in "Zelda" without getting "Elda". This can be problem for some people wanting to type in X/Z (user, password, settings, etc).

**Comment by i30817** [comment]
I'm still without directional keys due to this decision.

**Comment by lasers** [comment]
No worries. I wanted to let you know why obvious No. 2  isn't working. For other issues, I do not know and will leave it to others.

**Comment by i30817** [comment]
Ok. I'm not too mad about Z/X, because the alternatives are actually working on my keyboard, i just want to keep the possibility of 'menu remap' active for obvious reasons

**Comment by i30817** [comment]
BTW, how can you clear a mapping?

**Comment by lasers** [comment]
I hadn't played with remaps at all. I know you can use spacebar to reset the bind and the answer I can give right now is to find your config and clear them out manually. (This could be the only answer too).

**Comment by fr500** [comment]
Remaps or input bindings?
For remaps you just press start and it will reset the CORE<-->RetroPad mapping to default

For bindings, there is no way to clear them from the GUI

**Comment by i30817** [comment]
Bindings, but i did it from the retroarch.cfg file. I really think you should reserve a key, possibly pressed and not released for a few second to clear them completely. 
On a 'good news' the keybindings that didn't work - bug 4) in the OP - were able to be inputed from the cfg file, although i haven't tested if they actually work in game yet.

For the actual menu arrow keys, if remapping is too much to ask I'd be 'satisfied' if 
1. the arrow keys substitutes on the numpad actually worked on the menu 
2. that when the game actually starts, remaps that use the numpad are treated as singular keys (so no keypad 5 is equal to 5) and i'm suddenly able to control directions by pressing "1"
3. I'd _prefer_ if numlock and capslock status was irrelevant for game-level keybindings but i know how messy that can turn when the game or emulator requires text input. If that is impractical i want me pressing 'numlock' to actually change mode to work to enable the 'emulator keys' if i have to have one set for the menu and another for the emulator.

**Comment by i30817** [comment]
setting up the keybinds that don't work directly on the cfg file ( 4) in the OP ) works for some (divide,7,8 and 9 formed a correct axis on libretro unstable) but not on others 

5,1,2,3 as a left analog stick makes 5 correctly go up, 2 go left somehow (input_player1_l_y_plus = "keypad2") and 1 and 3 do nothing.

**Comment by MilesTailsFan2005** [comment]
use a newer retroarch config and copy the input text to the emuvr retroarch

**Comment by MilesTailsFan2005** [comment]
[retroarchconfig.txt](https://github.com/libretro/RetroArch/files/11072678/retroarchconfig.txt)

heres myn

**Comment by zoltanvb** [comment]
This is a rather old bug. Testing the situation now in the Remote Retropad keyboard tester, the button presses are recognized irrespective of the NumLock state.

**Comment by hizzlekizzle** [comment]
yeah, nothing we can do about emuVR, either, since it's stuck on an ooooold version.

Sounds like we're good here--or as good as we can get--so I'll close it out. We can reopen it if needed.
